src/main/webapp/app/shared/sidebar/sidebar-accordion/sidebar-accordion.component.ts

```diff
@@ -93,4 +104,84 @@ export class SidebarAccordionComponent implements OnChanges, OnInit {
         this.collapseState[groupCategoryKey] = !this.collapseState[groupCategoryKey];
         localStorage.setItem('sidebar.accordion.collapseState.' + this.storageId + '.byCourse.' + this.courseId, JSON.stringify(this.collapseState));
     }
+
+    private getWeekKey(date: dayjs.Dayjs): string {
+        const weekStart = date.startOf('isoWeek');
+        const weekEnd = date.endOf('isoWeek');
+        const year = weekStart.year();
+        return `${year} - ${weekStart.format('DD MMM YYYY')} - ${weekEnd.format('DD MMM YYYY')}`;
+    }
+
+    getGroupedByWeek(groupKey: string): WeekGroup[] {
```

Comment:
please move this into a util file and make it static or a pure function. This allows to better test the functionality and make sure that all edge cases are covered